src/controllers/keyboard/keyboardeventfilter.cpp

```diff
@@ -38,39 +34,51 @@ bool KeyboardEventFilter::eventFilter(QObject*, QEvent* e) {
 
         // Run through list of active keys to see if the pressed key is already active
         // Just for returning true if we are consuming this key event
-
-        foreach (const KeyDownInformation& keyDownInfo, m_qActiveKeyList) {
+        for (const KeyDownInformation& keyDownInfo: m_qActiveKeyList) {
             if (keyDownInfo.keyId == keyId) {
                 return true;
             }
         }
 
-        QKeySequence ks = getKeySeq(ke);
+        if (CmdlineArgs::Instance().getDeveloper()) {
+            qDebug() << "keyboard press: " << QKeySequence(ke->key()).toString();
+        }
+
+        QString ks = getKeySeq(ke);
+        emit keyseqPressed(ks);
+
+        // If no key-mapping info is known, there is nothing to check
+        if (m_kbdPreset.isNull()) {
+            return false;
+        }
+
         if (!ks.isEmpty()) {
-            ConfigValueKbd ksv(ks);
             // Check if a shortcut is defined
             bool result = false;
-            // using const_iterator here is faster than QMultiHash::values()
-            for (QMultiHash<ConfigValueKbd, ConfigKey>::const_iterator it =
-                         m_keySequenceToControlHash.find(ksv);
-                 it != m_keySequenceToControlHash.end() && it.key() == ksv; ++it) {
-                const ConfigKey& configKey = it.value();
-                if (configKey.group != "[KeyboardShortcuts]") {
-                    ControlObject* control = ControlObject::getControl(configKey);
-                    if (control) {
-                        //qDebug() << configKey << "MIDI_NOTE_ON" << 1;
-                        // Add key to active key list
-                        m_qActiveKeyList.append(KeyDownInformation(
-                            keyId, ke->modifiers(), control));
-                        // Since setting the value might cause us to go down
-                        // a route that would eventually clear the active
-                        // key list, do that last.
-                        control->setValueFromMidi(MIDI_NOTE_ON, 1);
-                        result = true;
-                    } else {
-                        qDebug() << "Warning: Keyboard key is configured for nonexistent control:"
-                                 << configKey.group << configKey.item;
-                    }
+
+            // NOTE: Using const_iterator here is faster than QMultiHash::values()
+            QMultiHash<QString, ConfigKey> mapping = m_kbdPreset->m_mapping;
+            QMultiHash<QString, ConfigKey>::const_iterator iterator;
+            for (iterator = mapping.find(ks); iterator != mapping.end(); ++iterator) {
+
+                // NOTE: We are not breaking here because there could
+                // potentially be another action mapped to the same
+                // key sequence further down the hash table
+                if (iterator.key() != ks) continue;
```

Comment:
Please use a separate line and curly braces for the body of an if statement, even in those cases